src/components/structures/auth/Registration.js

```diff
@@ -352,14 +363,25 @@ module.exports = React.createClass({
         });
     },
 
-    _makeRegisterRequest: function(auth) {
+    _makeRegisterRequest: function(auth, inhibitLogin) {
+        // default is to inhibit login if we're trying to register with an email address
+        // We do this so that the client that gets spawned when clicking on the email
+        // verification link doesn't get logged in (it can't choose different params
+        // because it doesn't have the password and it can only supply a complete
+        // set of parameters). If the original client is still around when the
+        // registration completes, it can resubmit with inhibitLogin=false to
+        // log itself in!
+        if (inhibitLogin === undefined) inhibitLogin = Boolean(this.state.formVals.email);
+
         // Only send the bind params if we're sending username / pw params
         // (Since we need to send no params at all to use the ones saved in the
         // session).
         const bindThreepids = this.state.formVals.password ? {
             email: true,
             msisdn: true,
         } : {};
+        // Likewise inhibitLogin
+        if (!this.state.formVals.password) inhibitLogin = null;
```

Comment:
Looking at the render method, I guess this means the client that was opened from the e-mail. If so, could you clarify in the comment above? Also, I thought that if you also had to click "Log in" in the browser that was opened from the e-mail? Probably misunderstanding something ...

Reply:
ah yes, sorry - this was from when I was going to put a button in the original client to re-submit the request, which we've now decided not to do.